pam_tcb/pam_unix_acct.c

```diff
@@ -76,6 +76,35 @@ static int acct_shadow(unused pam_handle_t *pamh, const void *void_user)
 	return ACCT_SUCCESS;
 }
 
+/*
+ * Use an external helper binary to perform account management.
+ */
+static int run_chkpwd_binary(const char *user)
+{
+	char *argv[] = { CHKPWD_HELPER, "chkacct", NULL };
+	char config[8];
+	int retval_helper;
+
+	if (!pam_unix_param.helper)
+		goto end;
+
+	if (on(UNIX_SHADOW)) {
+		memcpy(config, "shadow\0\0", 8);
+	} else {
+		memcpy(config, "passwd\0\0", 8);
+	}
```

Comment:
Sorry, I don't understand this `on(UNIX_SHADOW)` check, given that pam_sm_acct_mgmt invokes `set(UNIX_SHADOW)` unconditionally.